htdocs/product/class/product.class.php

```diff
@@ -5293,7 +5293,7 @@ public function load_virtual_stock($includedraftpoforvirtual = null)
 		}
 		if (((!empty($conf->fournisseur->enabled) && empty($conf->global->MAIN_USE_NEW_SUPPLIERMOD)) || !empty($conf->supplier_order->enabled) || !empty($conf->supplier_invoice->enabled)) && empty($conf->reception->enabled)) {
 			// Case module reception is not used
-			$filterStatus = '4';
+			$filterStatus = empty($conf->global->SUPPLIER_ORDER_STATUS_FOR_VIRTUAL_STOCK) ? '4' : $conf->global->SUPPLIER_ORDER_STATUS_FOR_VIRTUAL_STOCK;
```

Comment:
As the name says the option "SUPPLIER_ORDER_STATUS_FOR_VIRTUAL_STOCK" is to define the status into table SUPPLIER_ORDER to take into account.
Here, we will analyse the table llx_receptiondet_batch (the name in v20 or llx_commande_fournisseur_dispatch in v14). This is a different table with different status. So using the same variable name is not possible.